apps/src/blockly/addons/cdoXml.js

```diff
@@ -46,16 +47,37 @@ export default function initializeBlocklyXml(blocklyWrapper) {
    * @returns {Object[]} An array of objects containing the created blocks and their positions.
    */
   blocklyWrapper.Xml.domToBlockSpace = function (workspace, xml) {
-    const partitionedBlockElements = getPartitionedBlockElements(
+    const {prioritizedBlocks, remainingBlocks} = getSplitBlockElements(
       xml,
       PROCEDURE_DEFINITION_TYPES
     );
     const blocks = [];
+
+    for (const xmlFunctionBlock of prioritizedBlocks) {
+      // Try to find the function name of the procedure definition block.
+      // If we find it, create a procedure model and add it to the map.
+      let functionName = xmlFunctionBlock.children[1].textContent;
+      if (functionName) {
+        while (Blockly.Procedures.isNameUsed(functionName, workspace)) {
```

Comment:
We've got similar logic here: https://github.com/code-dot-org/code-dot-org/blob/69877eb1ebd1033185edcfab4054d03ddc38f071/apps/src/blockly/addons/functionEditor.js#L185

Could we share code/extract this into a helper?